Build instructions are not up to date

I tried building with a fresh, clean installation of ROS Noetic using our documentation and it's not great.

Fix the instructions such that they make sense and work.

---

Observations so far:

- [x] `install_proto3.sh`:
  - not needed on Noetic & Melodic, comes with rosdep. Potentially harmful there. Only recommend for Xenial.

- [x] Ceres:
  - installed by rosdep, no need to have it in the ws for Noetic (rosinstall). Do we need it for Xenial? --> fixed via #1532 

- [x] Abseil:
  - not mentioned
  - `install_abseil.sh` is provided by cartographer - installation to a catkin build dir would be nice, but the script uses `stow` so it can be "uninstalled" with `stow -D absl`. Should be fine if we explain what happens.
  - **todo**: mention installation of stow

- [x] this is required: https://github.com/cartographer-project/cartographer/pull/1705/ (the fix in FindGmock) ---> merged

Thank you, I face a build problem on ROS Noetic by following the official guide.
`CMake Error: The following variables are used in this project, but they are set to NOTFOUND.`

By installing the Abseil (`install_abseil.sh`) and the following cmake file change:
>this is required: cartographer-project/cartographer#1705 (the fix in FindGmock)

It works. Thank you again.

--- a/docs/source/compilation.rst
+++ b/docs/source/compilation.rst
@@ -42,14 +42,14 @@ On Ubuntu Focal with ROS Noetic use these commands to install the above tools:
 .. code-block:: bash
 
     sudo apt-get update
-    sudo apt-get install -y python3-wstool python3-rosdep ninja-build
+    sudo apt-get install -y python3-wstool python3-rosdep ninja-build stow
 
 On older distributions:
 
 .. code-block:: bash
 
     sudo apt-get update
-    sudo apt-get install -y python-wstool python-rosdep ninja-build
+    sudo apt-get install -y python-wstool python-rosdep ninja-build stow
 
 After the tools are installed, create a new cartographer_ros workspace in 'catkin_ws'.
 
@@ -61,20 +61,25 @@ After the tools are installed, create a new cartographer_ros workspace in 'catki
     wstool merge -t src https://raw.githubusercontent.com/cartographer-project/cartographer_ros/master/cartographer_ros.rosinstall
     wstool update -t src
 
-Install cartographer_ros' dependencies (proto3 and deb packages).
+Now you need to install cartographer_ros' dependencies.
+First, we use ``rosdep`` to install the required packages.
 The command 'sudo rosdep init' will print an error if you have already executed it since installing ROS. This error can be ignored.
 
 .. code-block:: bash
 
-    src/cartographer/scripts/install_proto3.sh 
     sudo rosdep init
     rosdep update
     rosdep install --from-paths src --ignore-src --rosdistro=${ROS_DISTRO} -y
 
-This builds Cartographer from the latest HEAD of the master branch.
-If you want a specific version, you need to change the version in the cartographer_ros.rosinstall.
+    # Only on Ubuntu 16 / ROS Kinetic: src/cartographer/scripts/install_proto3.sh 
 
-Additionally, uninstall the ros abseil-cpp using
+Cartographer uses the `abseil-cpp`_ library that needs to be manually installed using this script:
+
+.. code-block:: bash
+
+    src/cartographer/scripts/install_abseil.sh 
+
+Due to conflicting versions you might need to uninstall the ROS abseil-cpp using
 
 .. code-block:: bash
 
@@ -85,3 +90,8 @@ Build and install.
 .. code-block:: bash
 
     catkin_make_isolated --install --use-ninja
+
+This builds Cartographer from the latest HEAD of the master branch.
+If you want a specific version, you need to change the version in the cartographer_ros.rosinstall.
+
+.. _abseil-cpp: https://abseil.io/

